### DISCUSSION TITLE
How can I create a top level table of contents listing chapters, and sub- table of contents for sections in each chapter?

### DISCUSSION THREAD

**Discussion by daviewales** [quarto-dev/quarto-cli#7026] [Q&A]
### Description

I have a default Quarto book, created using 'new project > quarto book' in R Studio.

I want to include high level table of contents on the first page, listing each chapter.
Then, at the start of each chapter, I want a sub-table of contents, listing all the sections within that chapter.
I want this to appear in the Word docx output.

Is this possible in Quarto?

I have tried inserting `toc: true` at the start of each chapter file, but I only see a single TOC at the start of the document.

I have attached my test project:
[test-quarto-book.zip](https://github.com/quarto-dev/quarto-cli/files/12733050/test-quarto-book.zip)

**EDIT**:

I can get close to what I'm looking for if I render the chapters individually. For example:

```
quarto render summary.qmd --to docx
```

Note that I have to temporarily remove the `_quarto.yml` file to make this work. Otherwise it renders the whole book. Is this intended?

My goal is to render the first page based on the chapters, similar to how the book's current first page is rendered, with:

```
toc: true
toc-depth: 1
```

Then render each chapter similar to how they would be rendered stand-alone with:

```
toc: true
toc-depth: 3
```

Then concatenate all of these into a single word docx document. (PDF would be good too.)

**Comment by daviewales** [comment]
I'm guessing this might require a [pandoc filter](https://quarto.org/docs/extensions/filters.html)?

**Reply by mcanouil** [reply]
Indeed, possibly a good starting point would be the "section bibliographies" extension which has a very similar intent, see <https://github.com/pandoc-ext/section-bibliographies>.

Also relevant to you:
- https://stackoverflow.com/questions/60008898/pandoc-separate-table-of-contents-for-each-section
